npm/private/npm_import.bzl

```diff
@@ -394,14 +392,27 @@ def _download_and_extract_archive(rctx):
                 "password": rctx.attr.npm_auth,
             },
         }
-    elif rctx.attr.npm_auth_username and rctx.attr.npm_auth_password:
+        auth_count += 1
+    if rctx.attr.npm_auth_basic:
+        auth = {
+            download_url: {
+                "type": "pattern",
+                "pattern": "Basic <password>",
+                "password": rctx.attr.npm_auth_basic,
+            },
+        }
+        auth_count += 1
+    if rctx.attr.npm_auth_username and rctx.attr.npm_auth_password:
```

Comment:
nit: I think you're missing a check for e.g. npm_auth_basic + npm_apth_username (but no _password)

Reply:
that isn't valid. npm_auth_basic is the base64 encoded `username:password` so it stands on its own

Reply:
there is also a check above that username & password always travel together:
```
    if rctx.attr.npm_auth_username or rctx.attr.npm_auth_password:
        if not rctx.attr.npm_auth_username:
            fail("'npm_auth_password' was provided without 'npm_auth_username'")
        if not rctx.attr.npm_auth_password:
            fail("'npm_auth_username' was provided without 'npm_auth_password'")
```